basic-auth-sidecar/nginx.conf

```
@@ -20,7 +20,8 @@ http {
 
     log_format  main  '$remote_addr - $remote_user [$time_local] "$request" '
                       '$status $body_bytes_sent "$http_referer" '
-                      '"$http_user_agent" "$http_x_forwarded_for"';
+                      '"$http_user_agent" "$http_x_forwarded_for"'
```

should this log line have a white space character at the end like the above?
```suggestion
                      '"$http_user_agent" "$http_x_forwarded_for" '
```

---

No, not this one. These lines are concatenated as-is, ie.

```
'"hello" "world" '
'"my" "name" "is" "Tom"'

=>

'"hello" "world" "my" "name" "is" "Tom"'
```

If the last line had a trailing space, the log line would be output with a trailing space:

```
'"hello" "world" '
'"my" "name" "is" "Tom" '

=>

'"hello" "world" "my" "name" "is" "Tom" '
```

Which is undesirable!

---

Is that still true considering line 24 adds `'"http_cloudfront_viewer_address"'` to the end?

---

🤦🏻‍♂️ ☠️